File: Remove Element - Leetcode 27/Remove Element - Leetcode 27.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        i = 0
        n = len(nums)

        while i < n:
            if nums[i] == val:
                nums[i] = nums[n - 1]
                n -= 1
            else:
                i += 1
        return n

        # Time: O(n)
        # Space: O(1)

File: Remove Element - Leetcode 27/Remove Element - Leetcode 27.js
var removeElement = function(nums, val) {
    let i = 0;
    let n = nums.length;

    while (i < n) {
        if (nums[i] === val) {
            nums[i] = nums[n - 1];
            n--;
        } else {
            i++;
        }
    }
    return n;
};
